src/transformers/models/vit_mae/modeling_vit_mae.py

```diff
@@ -892,17 +982,21 @@ def unpatchify(self, patchified_pixel_values):
                 Pixel values.
         """
         patch_size, num_channels = self.config.patch_size, self.config.num_channels
-        num_patches_one_direction = int(patchified_pixel_values.shape[1] ** 0.5)
+        if not (original_img[2] == original_img[3] == 244):
+            num_patches_one_direction = original_img[2] // patch_size
+            num_patches_two_direction = original_img[3] // patch_size
+        else:
+            num_patches_one_direction = num_patches_two_direction = int(patchified_pixel_values.shape[1] ** 0.5)
```

Comment:
We can just use the first case, without having the do the conditional

```suggestion
        original_height, original_width = original_image_size
        num_patches_h = original_height // patch_size
        num_patches_w = original_width // patch_size
```